### PR TITLE
[autoscaler] Hide `accelerator_type` resources in `ray status`

### LINKED ISSUE #33272: [Ray status] confusing output about gpus and accelerators
### What happened + What you expected to happen

1. it's not clear it means 5 gpus + 5 T4s or 5 gpus which are T4s.
Screen Shot 2023-03-13 at 2 57 34 PM

2. The answer from 1 ought to be 5 gpus which are T4s. However, the output is not consistent. When gpu is used, the t4 is still not used.
```
Usage:
40.0/40.0 CPU (21.0 used of 21.0 reserved in placement groups)
5.0/5.0 GPU (5.0 used of 5.0 reserved in placement groups)
0.0/5.0 accelerator_type:T4
0B/101.35GiB memory
215.40MiB/44.70GiB object_store_memory
```

### Versions / Dependencies

ray-ml:nightly
py310

### Reproduction script

Start a cluster with those nodes and run something to use them.

### Issue Severity

Low: It annoys or frustrates me.

### PR DESCRIPTION
**Pull request by wuisawesome**
## Why are these changes needed?

It seems the `accelerator_type` field in `ray status was confusing people. This PR stops us from displaying it (except in verbose mode, where the user is expected to be more advanced and benefit from seeing a more advanced cluster state similar to how the autoscaler sees it. 

## Related issue number
Partially addresses #33272

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by wuisawesome** [comment]
cc @rkooo567 @scottsun94 @gvspraveen

**Review by ericl — APPROVED** [review]

**Comment by scottsun94** [comment]
@wuisawesome 
Will the second issue mentioned in https://github.com/ray-project/ray/issues/33272 be addressed? It will still happen in verbose mode?

**Comment by wuisawesome** [comment]
Only the GPU usage will be shown. The user can infer the type of GPU based on the node type they launched. 

Accurately figuring out which types of GPUs are being used would require significant redesign (if you feel strongly, feel free to bring it up in a planning meeting), though I'm open to making it easier to correlate node ip with node type in verbose mode in the future.

**Comment by scottsun94** [comment]
Sorry. I didn't quite follow. IIUC, after this PR is merged, here is how it might look:
in normal mode:
> Usage:
40.0/40.0 CPU (21.0 used of 21.0 reserved in placement groups)
5.0/5.0 GPU (5.0 used of 5.0 reserved in placement groups)
0B/101.35GiB memory
215.40MiB/44.70GiB object_store_memory

in verbose mode:
> Usage:
40.0/40.0 CPU (21.0 used of 21.0 reserved in placement groups)
**5.0/5.0 GPU (5.0 used of 5.0 reserved in placement groups)
0.0/5.0 accelerator_type:T4**
0B/101.35GiB memory
215.40MiB/44.70GiB object_store_memory


My question is: why is it "5.0/5.0 GPU" and "0.0/5.0 accelerator_type:T4"? This seems inconsistent. Is it a bug or something or is my misinterpretation?

**Comment by wuisawesome** [comment]
Yup that's correct, because there's a custom resource called `accelerator_type:T4` that is unused, which the scheduler and autoscaler use to make scheduling decisions. It may seem inconsistent or confusing, but it _is_ accurate. 

It's important for my CUJ (debugging downscaling) that the actual resource usage is shown. Hiding confusing, but true things would break my CUJ. 

Redesigning the `accelerator_type` feature is a fair request for @scv119 and the core team but out of scope for this PR.

**Comment by wuisawesome** [comment]
The autoscaler tests are passing and the other failures are a known unrelated breakage in master.

@scottsun94 i'm going to merge this now, but we can leave the issue open if you want to find a less confusing way of implementing or conveying the `accelerator_type` field.

**Pull request merged**
